WebInstall/explainer_cross_domain.md

```diff
@@ -0,0 +1,259 @@
+
+```javascript
+/* example of querying for the state of an installation permission using the Permission API  */
+
+const { state } = await navigator.permissions.query({
+  name: "installation"
+});
+switch (state) {
+  case "granted":
+    navigator.install('https://elk.zone');
+    break;
+  case "prompt":
+    //shows the install button in the web UI
+    showInstallButton();
+    break;
+  case "denied":
+    redirectToAppStore();
+    break;
+}
+```
+####  **Install Sources manifest field**
+* A new field called `install_sources` will be added to the manifest file to have a control list of sites that can install the app. In its most restrictive case, the developer can specify to not allow installation from any other origin, in which case the PWA conforms to its usual behaviour of only being able to be installed from its same origin.
+
+    * **`getInstalledApps()` method: If supported by the UA, the `getInstalledApps` method returns a list of the content that has been installed from that installation origin.** This means the installation origin will be able to know which applications it has installed, until cache is cleared. The installation origin *will not* be informed of any apps installed by other means, whether via another installation origin, directly through the browser, or by a native app store. The method returns a list of manifest ids of content installed from the calling origin. Additionally, if the browser has an active 'Do Not Track (DNT)', equivalent 'Global Privacy Control (GPC)' setting, is in Private browsing mode, or is an opinionated browser towards privacy, this is ignored and installation origins will not be allowed to know if that application is installed.
+
+```json
+{
+    "name": "Awesome PWA",
+    "display": "standalone",
+    "start_url": "/index.html",
+    "install_sources": [ 
+	    {"origin": "apps.microsoft.com"},
+	    {"origin": "store.app"}
+    ]
+}
+```
+
+This new manifest field will protect the app from being listed in undesirable repositories and give the developer absolute control about where do they want the PWA to be installed from. At best, the developer can allow the PWA to be installed from any site ("`*`"), at its most restrictive, it can only allow installing from the app's same scope. This field is only for the JS API and does not interfere with existing ways of installing PWAs through mechanisms like enterprise policies.
+
+If no `install_sources` are present in the manifest file, the default should be to not allow an app to be installed from cross-origin sites.
+
+#### **Gating capability behind installation**
+A UA may choose to gate the `navigator.install` capability behind installation of the web content. This would serve as an additional trust signal from the user towards enabling the functionality.
```

Comment:
```suggestion
A UA may choose to gate the `navigator.install` capability behind a requirement that the installation origin itself is installed. This would serve as an additional trust signal from the user towards enabling the functionality.
```